osbrain/address.py

```diff
@@ -282,4 +321,6 @@ def twin(self):
         """
         kind = self.kind.twin()
         role = self.role.twin()
-        return self.__class__(self.transport, self.address, kind, role)
+        serializer = self.serializer.serializer_type
+        return self.__class__(self.transport, self.address, kind, role,
+                              serializer)
```

Comment:
If you implement `AgentAddressSerializer` as `str`, then you can simply do:

    return self.__class__(self.transport, self.address, kind, role,
                          self.serializer)